lang/java/reef-tang/tang/src/main/java/org/apache/reef/tang/formats/AvroClassHierarchySerializer.java

```diff
@@ -0,0 +1,255 @@
+
+    return AvroNode.newBuilder()
+            .setName(name)
+            .setFullName(fullName)
+            .setNamedParameterNode(AvroNamedParameterNode.newBuilder()
+                    .setSimpleArgClassName(simpleArgClassName)
+                    .setFullArgClassName(fullArgClassName)
+                    .setIsSet(isSet)
+                    .setIsList(isList)
+                    .setDocumentation(documentation)
+                    .setShortName(shortName)
+                    .setInstanceDefault(instanceDefault)
+                    .build())
+            .setChildren(children).build();
+  }
+
+  private static AvroNode newPackageNode(final String name,
+                                         final String fullName,
+                                         final List<AvroNode> children) {
+    return AvroNode.newBuilder()
+            .setPackageNode(AvroPackageNode.newBuilder().build())
+            .setName(name).setFullName(fullName).setChildren(children).build();
+  }
+
+  private static AvroConstructorArg newConstructorArg(final String fullArgClassName,
+                                                      final String namedParameterName,
+                                                      final boolean isFuture) {
+    return AvroConstructorArg.newBuilder()
+            .setFullArgClassName(fullArgClassName)
+            .setIsInjectionFuture(isFuture)
+            .setNamedParameterName(namedParameterName).build();
+  }
+
+  private static AvroConstructorDef newConstructorDef(final ConstructorDef<?> def) {
+    final List<AvroConstructorArg> args = new ArrayList<>();
+    for (final ConstructorArg arg : def.getArgs()) {
+      args.add(newConstructorArg(arg.getType(), arg.getNamedParameterName(), arg.isInjectionFuture()));
+    }
+    return AvroConstructorDef.newBuilder()
+            .setFullArgClassName(def.getClassName())
+            .setConstructorArg(args).build();
+  }
+
+  @Override
+  public void toFile(final ClassHierarchy classHierarchy, final File file) throws IOException {
+    final AvroNode avroNode = toAvro(classHierarchy);
+    final DatumWriter<AvroNode> avroNodeWriter = new SpecificDatumWriter<>(AvroNode.class);
+    try (final DataFileWriter<AvroNode> dataFileWriter = new DataFileWriter<>(avroNodeWriter)) {
+      dataFileWriter.create(avroNode.getSchema(), file);
+      dataFileWriter.append(avroNode);
+    }
+  }
+
+  @Override
+  public byte[] toByteArray(final ClassHierarchy classHierarchy) throws IOException {
+    final DatumWriter<AvroNode> requestWriter = new SpecificDatumWriter<>(AvroNode.class);
+    try (final ByteArrayOutputStream out = new ByteArrayOutputStream()) {
+      final BinaryEncoder encoder = EncoderFactory.get().binaryEncoder(out, null);
+      requestWriter.write(toAvro(classHierarchy), encoder);
+      encoder.flush();
+      return out.toByteArray();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
```

Comment:
I think it would be OK for these methods to throw `IOException` if the underlying operations fail.